File: analysis/2019-04-17/testDten.py
#!/usr/local/bin/python3
import os
import re
import synapseclient
from synapseclient import *
import pandas as pd



syn = synapseclient.Synapse()
syn.login()

#first get dataset

fpath = syn.get('syn18408380').path
tab = pd.read_csv(fpath)
rtab = tab[['specimenID','tumorType','totalCounts','Symbol']]
rtab.columns = ['sample','conditions','counts','gene']
rtab.to_csv('newTab.csv')


conds = ['Low Grade Glioma','Cutaneous Neurofibroma','Plexiform Neurofibroma']
#conds = ['Plexiform Neurofibroma']
allnets=[]

for condition in conds:
    mv=re.sub(' ','',condition)+'Prots.tsv'
    #run metaviper
    cmd="Rscript ../../../drug-target-expression-network/bin/runMetaViper.R --input newTab.csv --output "+mv+' --idtype hugo --condition \"'+condition+'\"'
    print(cmd)
    os.system(cmd)

    #then run pcsf
    mo=re.sub(' ','',condition)+'net.rds'
    allnets.append(mo)
    cmd = "Rscript ../../../drug-target-expression-network/bin/runNetworkFromGenes.R --input "+mv+" --condition \""+condition+"\" --b 3 --mu 5e-05 --w 2 --output "+mo
    print(cmd)
    os.system(cmd)

#now with all three run the meta-analysis
newcmd='Rscript ../../../drug-target-expression-network/bin/metaNetworkComparisons.R --input '+','.join(allnets)
print(newcmd)
os.system(newcmd)

File: analysis/2019-04-19/testDten.R

require(synapser)
synLogin()
require(parallel)

#first get dataset

fpath = synGet('syn18408380')$path
tab = read.csv(fpath,header=T)
rtab = tab[,c('specimenID','tumorType','totalCounts','Symbol')]
colnames(rtab)<-c('sample','conditions','counts','gene')
#rtab.to_csv('newTab.csv')
write.csv(rtab,file='newTab.csv')

conds = c('Low Grade Glioma','Cutaneous Neurofibroma','Plexiform Neurofibroma')
allnets=c()


mclapply(conds,function(condition){
    mv=paste(gsub(' ','',condition),'Prots.tsv',sep='')
    #run metaviper
    cmd=paste("Rscript /usr/local/bin/runMetaViper.R --input newTab.csv --output",mv,'--idtype hugo --condition \"',condition,'\"')
    print(cmd)
    system(cmd)

    #then run pcsf
    mo=paste(gsub(' ','',condition),'net.rds',sep='')
    allnets.append(mo)
    cmd = paste("Rscript /usr/local/bin/runNetworkFromGenes.R --input",mv," --condition \"",condition,"\" --b 3 --mu 5e-05 --w 2 --output",mo)
    print(cmd)
    system(cmd)
})
#now with all three run the meta-analysis
newcmd=paste('Rscript /usr/local/bin/metaNetworkComparisons.R --input ',paste(allnets,collapse=','))
print(newcmd)
system(newcmd)
